### PR TITLE
Cypress/E2E: Upgrade cypress to v5.1.0

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Upgraded Cypress to v5.1.0.
- removed `testWithConfig` after getting an error "... detected you registered a(n) 'afterAll' hook while a test was running...". Our global before hook already handled reverting change on system config, so it's fine not to do it on per-spec file basis.
- replaced `merge-deep` with `deepmerge` after getting an error "Cannot find module 'kind-of'". Such error seemed to be from the dependency of `merge-deep`. Went ahead with `deepmerge` since it just do the same but without other dependency (so more stable).
- replaced `be` and `have` chainers with `equal` and `contain`

Test report: https://mm-cypress-report.s3.amazonaws.com/961-cypress-510-master/mochawesome.html (same result as daily prod test)

#### Ticket Link
none

### CODE REVIEW

File: e2e/cypress/integration/websocket/channel_created/new_sidebar_spec.js [hunk 1]
```diff
@@ -10,15 +10,15 @@
...
-    testWithConfig({
-        ServiceSettings: {
-            ExperimentalChannelSidebarOrganization: 'default_on',
-        },
+    before(() => {
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                ExperimentalChannelSidebarOrganization: 'default_on',
+            },
+        });
```
Comment (saturninoabril) [inline e2e/cypress/integration/websocket/channel_created/new_sidebar_spec.js:21]: Related Cypress error
Screen Shot 2020-09-15 at 1 20 55 AM

File: e2e/cypress/integration/messaging/message_deletion_spec.js [hunk 2]
```diff
@@ -43,10 +43,10 @@ describe('Message deletion', () => {
                 cy.get('#deletePostModal').should('be.visible');
 
                 // * Check that confirmation dialog contains correct text
-                cy.get('#deletePostModal').should('have', 'Are you sure you want to delete this Post?');
+                cy.get('#deletePostModal').should('contain', 'Are you sure you want to delete this Post?');
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/message_deletion_spec.js:46]: Related Cypress error
Screen Shot 2020-09-15 at 6 44 37 PM

File: e2e/cypress/integration/messaging/edit_message_spec.js [hunk 3]
```diff
@@ -114,7 +114,7 @@ describe('Edit Message', () => {
                 cy.get('#rhsContainer').should('be.visible');
 
                 // * Check that the RHS timeStamp equals the original post timeStamp
-                cy.get(`#CENTER_time_${postId}`).find('time').invoke('attr', 'dateTime').should('be', originalTimeStamp);
+                cy.get(`#CENTER_time_${postId}`).find('time').invoke('attr', 'dateTime').should('equal', originalTimeStamp);
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/edit_message_spec.js:117]: Related Cypress error
Screen Shot 2020-09-15 at 8 36 10 PM

**Review by hmhealey — APPROVED** [review]

**Review by srkgupta — APPROVED** [review]
Thanks for adding the final report @saturninoabril. The changes LGTM.

**Comment by saturninoabril** [comment]
Hello @sudheerDev, please take a look if you get a chance. I'm hoping to get this in before branching out with 5.28.  Thanks!

**Review by sudheerDev — APPROVED** [review]
🎉

**Pull request merged**